lib/html_cleaner.rb

```diff
@@ -141,13 +141,9 @@ def add_paragraphs_to_text(text)
   # These assume they are running on well-formed XHTML, which we can do
   # because they will only be used on already-cleaned fields.
 
-  # strip img tags, optionally leaving the src URL exposed
+  # strip img tags, optionally leaving the HTML attributes (e.g. src and alt) exposed
   def strip_images(value, keep_src: false)
-    value.gsub(/(<img .*?>)/) do |img_tag|
-      match_data = img_tag.match(/src="([^"]+)"/) if keep_src
-      src = match_data[1] if match_data
-      src || ""
-    end
+    value.gsub(%r{(?:<(img .*?) ?/?>)}, keep_src ? "\\1" : "")
```

Comment:
The way I understand the ` ?` before the trailing slash is that you're trying to remove the trailing space from the source. I think it would be clearer to instead `.strip` the `"\\1"`, since regex tends to be harder to read

Reply:
Not quite, and the difference means `.strip` won't work. This regex is intended to match
```html
<img src="foo" />
<img src="foo"/>
<img src="foo">
```
where both the space and trailing slash are optional. (Technically this also supports the case where `.strip` _would_ work -- `<img src="foo" >` -- but that's not really the primary goal.) They're specified now to avoid grabbing them in the inner (capture group (alongside the HTML attributes)

Reply:
I see, thank you for explaining!

Reply:
Won't this also make it so any allowed attributes that are basically just clutter when the image isn't being embedded (e.g., width or border) will still be left behind? Or am I missing something that makes it only apply to src and alt?

Reply:
It will. I decided that was an OK tradeoff to make the code much simpler than it was otherwise